acra/security-controls/zones.md

```diff
@@ -0,0 +1,146 @@
+
+### Decrypt with AcraServer
+
+To decrypt data AcraServer should know which ZoneID should use to decrypt data. It processes result rows from database and before any decryption tries to match ZoneID.
+Only after successful match ZoneID AcraServer starts matching AcraStruct/AcraBlock signatures in result data. 
+So, every ZoneID should be placed before appropriate piece of encrypted data in result rows.
+
+Because SQL is the only way how application communicates with database and AcraServer, there is only one way where AcraServer can get and match ZoneID in pair with encrypted data: database result rows.
+There are several methods how to achieve it:
+* ZoneID should be stored in database and precedes encrypted data in result rows
+* ZoneID specified explicitly in SQL query as literal (according to database binary literals) in `SELECT` statement before encrypted column
+* ZoneID specified explicitly in SQL query as literal (according to database binary literals) in `SELECT` statement before explicitly specified encrypted data as SQL literal
+
+#### Method 1
+
+Let's look how it works on first method on simple example when application stores ZoneID and encrypted data in database table. 
+For PostgreSQL will look like:
+```
+CREATE TABLE application_data (
+ id SERIAL PRIMARY KEY,
+ zone_id BYTEA,
+ data BYTEA
+);
+INSERT INTO application_data (zone_id, data) VALUES ('DDDDDDDDjKjECtcRBDkmHVBh'::BYTEA, '\x2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103'::BYTEA);
+```
+For MySQL will look like:
+```
+CREATE TABLE application_data (
+ id INTEGER PRIMARY KEY AUTO_INCREMENT,
+ zone_id BLOB,
+ data BLOB
+);
+INSERT INTO application_data (zone_id, data) VALUES ('DDDDDDDDjKjECtcRBDkmHVBh', X'2222222222222222554543320000002d116bab650305cf67209623ed3a134fd77bfecd0c9a95107450826e14f950fdd1dba73732872027042654000000000101400c00000010000000200000003f5fd06dbf8bf49be6a8b440ea54f01174934049fd563ce27ff0aafbe5ea9155588e1ddd0ce64804fe5ff347ae097e29dd007fcaa02a3548da568df83300000000000000000101400c00000010000000070000002273af944d98bcde697b914d98fea013b77a358a93959ddfee47858b75d2e86eb5f103');
+```
+Here we created table where ZoneID will be stored in `zone_id` and this column precedes the `data` column where stored encrypted data. So if we query `SELECT * FROM application_data` 
+it will be expanded by database to `SELECT "application_data"."id", "application_data"."zone_id", "application_data"."data"` and result rows will contain ZoneID before encrypted data.
+In our example result from PostgreSQL will be:
```

Comment:
okay, will change. what about quoting. it works similar for postgresql and MySQL, it's ansi standard. but okay, removed them.

Reply:
> what about quoting. it works similar for postgresql and MySQL, it's ansi standard

It is, but with MySQL you need to explicitly set `ANSI_QUOTES` mode for it to accept ANSI quotes instead of their default choice of backticks.